lib/api/checkr_api.py

```diff
@@ -0,0 +1,128 @@
+        phone: Optional[str] = None,
+        geo_ids: Optional[list[str]] = None,
+        previous_driver_license_number: Optional[str] = None,
+        previous_driver_license_state: Optional[str] = None,
+        copy_requested: Optional[bool] = None,
+        custom_id: Optional[str] = None,
+    ) -> str:
+        candidate_data = {
+            "first_name": first_name,
+            "last_name": last_name,
+            "email": email,
+            "driver_license_number": driver_license_number,
+            "driver_license_state": driver_license_state,
+            "work_locations": work_locations,
+            "middle_name": middle_name,
+            "mother_maiden_name": mother_maiden_name,
+            "ssn": ssn,
+            "dob": dob,
+            "zipcode": zipcode,
+            "uri": uri,
+            "phone": phone,
+            "geo_ids": geo_ids,
+            "previous_driver_license_number": previous_driver_license_number,
+            "previous_driver_license_state": previous_driver_license_state,
+            "copy_requested": copy_requested,
+            "custom_id": custom_id
+        }
+
+        candidate_json = json.dumps(candidate_data)
+        response = requests.post(
+            auth=HTTPBasicAuth(self.secret_api_key, None),
+            url=f"{self.api_url}/candidates",
+            data=candidate_json
+        )
+
+        candidate = response.json()
+        return candidate
+
+    def create_invitation(
+        self,
+        candidate_id: str,
+        package: str,
+        work_locations: list[dict[str, str]] = None,
+        node: Optional[str] = None,
+        tags: Optional[str] = None,
+        communication_types=None,
+    ) -> dict[str, Any]:
+        invitation_data = {
+            "package": package,
+            "candidate_id": candidate_id,
+            "work_locations": work_locations,
+            "node": node,
+            "tags": tags,
+            "communication_types": communication_types
+        }
+
+        invitation_json = json.dumps(invitation_data)
+        response = self.session.post(
+            auth=HTTPBasicAuth(self.secret_api_key, None),
+            url=f"{self.api_url}/invitations",
+            data=invitation_json
+        )
+
+        invitation = response.json()
+        return invitation
+
+    def create_report(
+        self,
+        candidate_id: str,
+        package: str,
+        work_locations: list[dict[str, str]],
+        node: Optional[str] = None
+    ) -> dict[str, Any]:
+        report_data = {
+            "candidate_id": candidate_id,
+            "package": package,
+            "node": node,
+            "work_locations": work_locations,
+        }
+
+        report_json = json.dumps(report_data)
+        response = self.session.post(
+            auth=HTTPBasicAuth(self.secret_api_key, None),
+            url=f"{self.api_url}/reports",
+            data=report_json
+        )
+
+        report = response.json()
```

Comment:
created a local just to make debugging easier.

Reply:
Yeah, that's good practice. Self-documenting code is inherently more debuggable.